packages/vite/src/node/config.ts

```diff
@@ -434,7 +437,7 @@ export interface UserConfig extends DefaultEnvironmentOptions {
   /**
    * Environment overrides
    */
-  environments?: Record<string, EnvironmentOptions>
+  [key: `$${string}`]: EnvironmentOptions
```

Comment:
```suggestion
  [key: `$${string}`]: EnvironmentOptions | undefined
  $client?: EnvironmentOptions
  $ssr?: EnvironmentOptions
```
I guess this makes a bit easier to write env options.